src/Resource/GenericResource.php

```diff
@@ -244,4 +222,69 @@ protected function postUnserialize(array $data)
         $this->repoPath = array_pop($data);
         $this->path = array_pop($data);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function pathGetChild($path, $relPath)
+    {
+        if (!$this->getRepository()) {
+            throw ResourceNotFoundException::forPath($path.'/'.$relPath);
+        }
+
+        return $this->getRepository()->get($path.'/'.$relPath);
```

Comment:
I don't think these additional methods are necessary. Remember that method calls are relatively expensive in PHP. IMO it doesn't make sense to reduce code duplication by moving code to an extra method if that code is relatively short and trivial (like this code here).